src/hotspot/share/opto/predicates.cpp

```diff
@@ -1131,6 +1148,7 @@ void UpdateStrideForAssertionPredicates::visit(const TemplateAssertionPredicate&
     return;
   }
   replace_opaque_stride_input(template_assertion_predicate);
+  template_assertion_predicate.update_associated_loop_node(_loop_node);
```

Comment:
We don't need to clone the Template Assertion Expression and hence we only need to update the loop node.